flang/include/flang/Tools/CLOptions.inc

```diff
@@ -91,20 +98,27 @@ inline void addLLVMDialectToLLVMPass(
 inline void createMLIRToLLVMPassPipeline(mlir::PassManager &pm)
 {
   // simplify the IR
+  mlir::GreedyRewriteConfig config;
+  config.enableRegionSimplification = false;
   fir::addCSE(pm);
   pm.addNestedPass<mlir::FuncOp>(fir::createArrayValueCopyPass());
   pm.addNestedPass<mlir::FuncOp>(fir::createCharacterConversionPass());
-  pm.addPass(mlir::createCanonicalizerPass());
+  pm.addPass(mlir::createCanonicalizerPass(config));
```

Comment:
This change makes this function inconsistent with `bbc`. I think that it should be `pm.addPass(mlir::createCanonicalizerPass());` instead.

Reply:
The config is necessary here. That's the essence of this PR. Without the config the full canonicalizer is run.

Reply:
Why not in `bbc` then? This change is causing a regression in `flang-new` (and presumably in `tco` too).

Reply:
What regression are you talking about? `make check-flang` is fine. If it uses the CLOptions.inc there should be no regression since the correct pass is added ([pass ](https://github.com/clementval/llvm-project/blob/2490327bbcfe96a9b9accbd62fdcc76b985874b3/flang/include/flang/Tools/CLOptions.inc#L121))

Reply:
> What regression are you talking about? make check-flang is fine. 

There is no test in `fir-dev` that triggers this ATM. I added one in  #1281.

> Is flang-new not using the pipeline from CLOptions.inc?

It is. But `bbc` is currently **not using** `createMLIRToLLVMPassPipeline` from `CLOptions.inc` (and there's a discrepancy between `createMLIRToLLVMPassPipeline` and similar code in `bbc`). See #1280. Also - shall we move the discussion there?